### DISCUSSION TITLE
How to detect if a change was made

### DISCUSSION THREAD

**Discussion by Silfalion** [hm21/pro_image_editor#279] [Q&A] [answered]
Hello,

I'm trying to modify the crop and rotate editor to check if there were any changes before starting the edit as to not generate another image(and speeding up the last step for the user).

Right now I'm using the screenshot history variable, however this seems to be unreliable because the initial array has 1 element and the first editing replaces this element. 

Any ideas as to how to track it more reliably?

**Comment by hm21** [comment, marked as answer]
The “screenshot history” handles only the background-generated images. Therefore, in this case, you should access the “history” itself, which includes the class `TransformConfigs`. Within this class, you can call the `isEmpty` method to check whether there have been any changes to the transformations or if it remains empty. Below is an example:

```dart
var editor = editorKey.currentState!;
var cropEditor = editor.cropRotateEditor.currentState;
if (cropEditor?.history.isNotEmpty == true &&
    cropEditor!.history.first.isNotEmpty) {
  /// Your code
}
```

If you want to check in the main editor whether any changes have been applied from the crop-rotate editor, you can directly read the `transformConfigs` as shown below:

```dart
  if (editorKey.currentState!.stateHistory.last.transformConfigs.isNotEmpty) {
    /// Your code
  }
```
